### DISCUSSION TITLE
Not able to push code to github.

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#613] [General]
why these commands are not working and showing errors.
```
git config --global user.email"khushbusandhu12@gmail.com"
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git config --global user.name"khushbu"
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git commit -m 'FIRST COMMIT'
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.
```

**Comment by theirrationalone** [comment]
Actually you're not giving a space after user.email... 

user.email"khushbusandhu12@gmail.com" ❌
user.email "khushbusandhu12@gmail.com" ✅

And same for user.name too.... 


# A simple and recommended procedure to push a repo on github... 

Remote url: your github's repo https url
branch name: usually it has main Or master. 

```
git init

git add .

git commit -m "commit msg"

git remote add origin <remote-url>

git push origin <branch-name>

```

**Reply by khushbusandhu** [reply]
now vs code is asking for signing in to github but was not able to do so

**Reply by theirrationalone** [reply]
What's the error??

**Reply by khushbusandhu** [reply]
I am not getting anything about error . There is no error in terminal but after running the git push origin command there is nothing that worked

**Reply by khushbusandhu** [reply]
```
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git push -u origin main

``` 
this command actually is not red or green marked it means it is not able to run

**Reply by theirrationalone** [reply]
Have you done with this command... ?? 


```
git remote add origin <remote-url>
```

**Reply by theirrationalone** [reply]
Try to push with master instead of main.

**Reply by khushbusandhu** [reply]
yes

**Reply by theirrationalone** [reply]
In your repo there is no branch called main... 

So you can create one by using this command below. Or put master instead of main which is the default branch. 

```
git checkout -b main
```

**Reply by khushbusandhu** [reply]
actually after writing push command the terminal is not showing any response.
so I cannot be able to write anything after that

**Reply by khushbusandhu** [reply]
and this window got opened after running the last push command.
https://github.com/Visual-Studio-Code

**Reply by theirrationalone** [reply]
Would you like to share your repo url??

**Reply by khushbusandhu** [reply]
yes ! sure
https://github.com/khushbusandhu/foundry-fund-me-f23

**Reply by OMGitsNIK** [reply]
That repo appears to be empty, is the matter solved or would you like further assistance?

**Reply by khushbusandhu** [reply]
I am not getting your point.

**Reply by theirrationalone** [reply]
Hey can you please show me your commands with procedure you have executed?? 

I think it's a origin problem... 

Origin might not right there

**Reply by khushbusandhu** [reply]
here are all the commands from my terminal
```
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git config --global user.name "khushbu"
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git init -b main
Initialized empty Git repository in /home/khushbu/solidity-course/foundry-simple-storage-f23/fund-me/.git/
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git status
On branch main

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .gas-snapshot
        .github/
        .gitignore
        Makefile
        README.md
        foundry.toml
        script/
        src/
        test/

nothing added to commit but untracked files present (use "git add" to track)
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git add .
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git status
On branch main

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   .gas-snapshot
        new file:   .github/workflows/test.yml
        new file:   .gitignore
        new file:   Makefile
        new file:   README.md
        new file:   foundry.toml
        new file:   script/Deployfundme.s.sol
        new file:   script/HelperConfig.s.sol
        new file:   script/Interactions.s.sol
        new file:   src/PriceConverter.sol
        new file:   src/fundme.sol
        new file:   test/integration/InteractionsTest.t.sol
        new file:   test/mocks/MockV3Aggregator.sol
        new file:   test/unit/fundmeTest.t.sol

khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git commit -m 'FIRST COMMIT'
[main (root-commit) 913cc83] FIRST COMMIT
 14 files changed, 601 insertions(+)
 create mode 100644 .gas-snapshot
 create mode 100644 .github/workflows/test.yml
 create mode 100644 .gitignore
 create mode 100644 Makefile
 create mode 100644 README.md
 create mode 100644 foundry.toml
 create mode 100644 script/Deployfundme.s.sol
 create mode 100644 script/HelperConfig.s.sol
 create mode 100644 script/Interactions.s.sol
 create mode 100644 src/PriceConverter.sol
 create mode 100644 src/fundme.sol
 create mode 100644 test/integration/InteractionsTest.t.sol
 create mode 100644 test/mocks/MockV3Aggregator.sol
 create mode 100644 test/unit/fundmeTest.t.sol
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git branch -M main
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git remote add origin https://github.com/khushbusandhu/foundry-fund-me-f23.git
khushbu@LAPTOP-D6QBD1MV:~/solidity-course/foundry-simple-storage-f23/fund-me$ git push -u origin main

```
and last command do not responded

**Reply by theirrationalone** [reply]
you entered the url exactly like this below... ?? 

```
git remote add origin https://github.com/khushbusandhu/foundry-fund-me-f23
```

**Reply by khushbusandhu** [reply]
.git was there at the end

**Reply by theirrationalone** [reply]
First try to do push with master there could have some problems with main... 

If that doesn't work then reset origin url without  .git

**Reply by khushbusandhu** [reply]
did exactly the same but not working

**Reply by khushbusandhu** [reply]
everything is working correctly except the last command.  even that is not showing any error. it just not working and not showing any response

**Reply by theirrationalone** [reply]
Try this and verify the origin url

```
git remote -v
```

And please make sure your internet connection is ON 😅

**Reply by khushbusandhu** [reply]
```
origin  https://github.com/khushbusandhu/foundry-fund-me-f23.git (fetch)
origin  https://github.com/khushbusandhu/foundry-fund-me-f23.git (push)
```

here's the output

**Reply by theirrationalone** [reply]
Try to push again with --force subcommand

**Reply by khushbusandhu** [reply]
not working

**Reply by theirrationalone** [reply]
Try this one

```
git push --verbose origin main

```

**Reply by theirrationalone** [reply]
It might an internet connection problem

**Reply by khushbusandhu** [reply]
same result

**Reply by theirrationalone** [reply]
Some logs might be appeared on the terminal please share them with me

**Reply by khushbusandhu** [reply]
but i have decent internet connection

**Reply by khushbusandhu** [reply]
```
Pushing to https://github.com/khushbusandhu/foundry-fund-me-f23.git
``` 
this is what I got

**Reply by khushbusandhu** [reply]
but nothing after this

**Reply by theirrationalone** [reply]
> That repo appears to be empty, is the matter solved or would you like further assistance?

Yes please...

**Reply by theirrationalone** [reply]
Please follow this... Once again... 

git add .

git commit -m "FIRST COMMIT 🔥"


git push -u origin main


Please do these steps once again.

**Reply by khushbusandhu** [reply]
I have did these steps 10 times but same result

**Reply by theirrationalone** [reply]
Have you asked from chatGPT?? 

I can guess there could have permissions problem, origin problem.

**Reply by theirrationalone** [reply]
Final one thing we can do... 


Clone your own repo into a different folder.. Then copy paste this project into your newly cloned repo in your pc. Then try above step again. 

This one is last thing i often do...

**Reply by theirrationalone** [reply]
I reanalysed your logs the problem is here 

git config --global user.name "khushbu(here) "... Your username is khushbusandhu, not that 😅

By the way last reply will also work.

**Reply by khushbusandhu** [reply]
finally, it worked.

thanks, alot @theirrationalone I just can't thank you enough for all this help and for guiding me for so long. thanks a lot.

**Reply by theirrationalone** [reply]
We are the Community ! 😉

**Reply by khushbusandhu** [reply]
keep guiding beginners like me so that we can also get able to understand things better. since I am feeling like a noob right now I need that guidance 😅.

**Reply by theirrationalone** [reply]
I can see A PRO in You. 😉

**Reply by khushbusandhu** [reply]
This line made my day 😅

**Reply by khushbusandhu** [reply]
hey @theirrationalone !
can you provide me with some other resources that I can use to learn foundry and solidity effectively with more clear understanding?

**Reply by theirrationalone** [reply]
I often use Patrick's (SIR) all resources provided by him. 

Apart of it i'm following Johnny (SIR) [YOUTUBE](https://youtu.be/4JmqHnkSlFI) not limited to foundry & solidity but whole blockchain as like Patrick. 

Also share your resources too 😜

**Reply by shikhar229169** [reply]
hey @theirrationalone, have you completed the hardhat course?

**Reply by theirrationalone** [reply]
Yo 😋

**Reply by shikhar229169** [reply]
I need help, please provide your socials so I can contact you.

**Reply by theirrationalone** [reply]
This is my [twitter](https://twitter.com/theirrationalon?t=2EMQYM0dZ3FPRxZt45mUyA&s=09) handle but i hardly go there Or Anywhere else 😅

But for you i'm toggling my X's Notifications ON, i'll wait for your ping. Brother!

**Reply by shikhar229169** [reply]
Its not allowing me to message you.

**Reply by theirrationalone** [reply]
Send me yours

**Reply by shikhar229169** [reply]
[here](https://twitter.com/shikhar229)

**Reply by khushbusandhu** [reply]
hey @theirrationalone  
https://learnweb3.io/ 
This is the one course I am thinking of starting.

**Reply by theirrationalone** [reply]
> hey @theirrationalone  
> https://learnweb3.io/ 
> This is the one course I am thinking of starting.

Thanks @khushbusandhu , woww i got new world to explore and that's appeared interesting...... Thank You so much 🙏💕
